teensy40_brushed_v2/src/main_ros.cpp

```diff
@@ -0,0 +1,469 @@
+#include "common.h"
+#if USE_ROS_FIRMWARE == 1
+#include <Arduino.h>
+#include "hardware_pins.h"
+#include "driver_motor.h"
+#include "model_encoder.h"
+#include "model_sensor.h"
+#include "motor.h"
+
+#include <ros.h>
+#include "std_msgs/Float32.h"
+#include "std_msgs/Float32MultiArray.h"
+
+#include <memory>
+
+// CONFIGURATION
+#define PID_PERIOD_MS 100
+#define PID_PERIOD_US PID_PERIOD_MS * 1000
+#define HWSERIAL Serial2
+#define DEBUG_PRINT 1
+
+#define OUTA_Pin ENCPIN1_1
+#define OUTB_Pin ENCPIN1_2
+
+// ROS
+ros::NodeHandle nh;
+float arm_brushed_setpoint_ps[3] = {0, 0, 0};
+float arm_brushed_angle_ps[3] = {0, 0, 0};
+void brushed_board_ros_loop();
+void arm_brushed_cmd_cb(const std_msgs::Float32MultiArray &input_msg);
+std_msgs::Float32MultiArray arm_brushed_fb_msg;
+ros::Publisher arm_brushed_fb_pub("/armBrushedFb", &arm_brushed_fb_msg);
+ros::Subscriber<std_msgs::Float32MultiArray> arm_brushed_cmd_sub("/armBrushedCmd", arm_brushed_cmd_cb);
+
+// DUMB POINTERS
+driver_motor mot1;
+driver_motor mot2;
+driver_motor mot3;
+
+// TESTER VARIABLES
+int stage = 0;
+int pos = 90;
+int tolerance = 0.5;
+volatile boolean is_homed = false;
+
+// FUNCTION DECLARATIONS
+void brushed_board_homing();
+void brushed_board_ros_loop();
+
+void lim1ISR();
+void lim2ISR();
+void lim3ISR();
+void lim4ISR();
+void lim5ISR();
+void lim6ISR();
+
+// Define the size of the moving average window
+const uint32_t MOVING_AVERAGE_SIZE = 10;
+float cur1_voltage_buffer[MOVING_AVERAGE_SIZE] = {0};
+float cur2_voltage_buffer[MOVING_AVERAGE_SIZE] = {0};
+float cur3_voltage_buffer[MOVING_AVERAGE_SIZE] = {0};
+int cur1_voltage_buffer_idx = 0;
+int cur2_voltage_buffer_idx = 0;
+int cur3_voltage_buffer_idx = 0;
+
```

Comment:
In general, if there are a handful of function definitions and variable definitions, it would be fine putting them at the top of a .cpp file. Since there a lot of them, for better readability it would help putting them in a separate file.